File: app/src/main/java/eu/warble/pjapp/ui/login/LoginActivity.java
package eu.warble.pjapp.ui.login;


import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.design.widget.TextInputEditText;
import android.widget.Button;

import br.com.simplepass.loading_button_lib.customViews.CircularProgressButton;
import eu.warble.pjapp.R;
import eu.warble.pjapp.ui.MainActivity;
import eu.warble.pjapp.ui.base.BaseActivity;

public class LoginActivity extends BaseActivity<LoginPresenter> {

    private CircularProgressButton loginBtn;
    private Button loginAsGuestBtn;
    private TextInputEditText loginField, passwordField;

    @Override
    protected LoginPresenter createPresenter() {
        return new LoginPresenter(this);
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        setTheme(R.style.AppThemeWithoutActionBar);
        setContentView(R.layout.activity_login);
        initViews();
    }

    private void initViews() {
        loginBtn = findViewById(R.id.loginBtn);
        loginAsGuestBtn = findViewById(R.id.loginGuestBtn);
        loginField = findViewById(R.id.login);
        passwordField = findViewById(R.id.password);
        loginBtn.setOnClickListener(view -> login());
        loginAsGuestBtn.setOnClickListener(view -> startMainActivityWithGuestMode());
    }

    void showLoadingProgress(){
        loginBtn.startAnimation();
        loginField.setEnabled(false);
        passwordField.setEnabled(false);
        loginBtn.setEnabled(false);
        loginAsGuestBtn.setEnabled(false);
    }

    void hideLoadingProgress(){
        loginBtn.revertAnimation();
        loginField.setEnabled(true);
        passwordField.setEnabled(true);
        loginBtn.setEnabled(true);
        loginAsGuestBtn.setEnabled(true);
    }

    private void login(){
        loginBtn.setEnabled(false);
        showLoadingProgress();
        presenter.login(loginField.getText().toString(), passwordField.getText().toString());
    }

    public void showError(int stringResId){
        loginBtn.setEnabled(true);
        hideLoadingProgress();
        Snackbar.make(findViewById(R.id.loginMainLayout), getString(stringResId), Snackbar.LENGTH_SHORT)
                .show();
    }

    public void showError(String error){
        loginBtn.setEnabled(true);
        hideLoadingProgress();
        Snackbar.make(findViewById(R.id.loginMainLayout), error, Snackbar.LENGTH_SHORT)
                .show();
    }

    public void startMainActivity(){
        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
        finish();
    }

    public void startMainActivityWithGuestMode() {
        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        intent.putExtra("guestMode", true);
        startActivity(intent);
        finish();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        loginBtn.dispose();
    }
}

File: app/src/main/java/eu/warble/pjappkotlin/mvp/login/LoginActivity.kt
package eu.warble.pjappkotlin.mvp.login

import android.os.Bundle
import android.view.View
import eu.warble.pjappkotlin.R
import eu.warble.pjappkotlin.mvp.ApplicationNavigator
import eu.warble.pjappkotlin.mvp.BaseActivity
import kotlinx.android.synthetic.main.activity_login.*
import android.content.Intent
import android.net.Uri
import eu.warble.pjappkotlin.utils.Constants
import eu.warble.pjappkotlin.utils.Tools


class LoginActivity : BaseActivity(), LoginContract.View {

    override lateinit var presenter: LoginContract.Presenter
    override val applicationNavigator: ApplicationNavigator = ApplicationNavigator(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.AppThemeWithoutActionBar)
        setContentView(R.layout.activity_login)
        presenter = LoginPresenter(
            this,
            applicationContext
        )
        initViews()
    }

    private fun initViews() {
        loginBtn.setOnClickListener {
            presenter.login(login.text.toString(), password.text.toString())
        }

        loginGuestBtn.setOnClickListener {
            if (presenter.isNetworkAvailable())
                applicationNavigator.goToMainActivityWithGuestMode()
            else
                showError(R.string.connect_error)
        }

        val privacyPolicyText = privacy_policy
        privacyPolicyText.text = Tools.fromHtml(getString(R.string.privacy_policy_message))
        privacyPolicyText.setOnClickListener {
            openPrivacyPolicyPage()
        }
    }

    private fun openPrivacyPolicyPage() {
        applicationNavigator.openWebPage(Constants.PRIVACY_POLICY_URL)
    }

    override fun onResume() {
        super.onResume()
        presenter.start()
    }

    override fun showLoading(show: Boolean) {
        loading_screen.visibility = if (show) View.VISIBLE else View.GONE
        loginGuestBtn.isEnabled = !show
        loginBtn.isEnabled = !show
    }
}
